src/components/cc-heptapod-info/cc-heptapod-info.js

```diff
@@ -18,40 +18,43 @@ const HEPTAPOD_LOGO_URL = 'https://assets.clever-cloud.com/logos/heptapod.svg';
 
 /**
  * @typedef {import('./cc-heptapod-info.types.js').Statistics} Statistics
+ * @typedef {import('./cc-heptapod-info.types.js').HeptapodInfoState} HeptapodInfoState
+ * @typedef {import('./cc-heptapod-info.types.js').HeptapodInfoStateLoaded} HeptapodInfoStateLoaded
+ * @typedef {import('./cc-heptapod-info.types.js').HeptapodInfoStateLoading} HeptapodInfoStateLoading
  */
 
 /**
  * A component that shows a summary of our Heptapod SaaS offer.
  *
- * ## Details
- *
- * * When `statistics` is nullish, a skeleton screen UI pattern is displayed (loading hint).
- *
  * @cssdisplay block
  */
 export class CcHeptapodInfo extends LitElement {
 
   static get properties () {
     return {
-      error: { type: Boolean, reflect: true },
-      statistics: { type: Object },
+      state: { type: Object },
     };
   }
 
   constructor () {
     super();
 
-    /** @type {boolean} Displays an error message. */
-    this.error = false;
+    /** @type {HeptapodInfoState} Set the state of the component. */
+    this.state = { type: 'loading' };
+  }
 
-    /** @type {Statistics|null} Sets the usage statistics of this heptapod SaaS or `"not-used"` to display a message explaining the service is not used. */
-    this.statistics = null;
+  /**
+   * @param {HeptapodInfoState} state
+   * @return {state is (HeptapodInfoStateLoaded|HeptapodInfoStateLoading)}
+   * @private
+   */
+  _isLoadedOrLoading (state) {
+    return state.type === 'loaded' || state.type === 'loading';
   }
```

Comment:
nitpick: since it's a rather simple method and to improve "reading code from just one place", I'd rather move this to a variable inside the render function.

Reply:
You cannot inline this into a variable within the `render` function because this will break TypeScript inference (at least for now, with TS 5.5 it may become ok). For such cases you have to use a type predicate or find a smarter solution. 
The solution you suggest below is smarter so I'll see what @HeleneAmouzou thinks about it but we'll probably go for that, thanks :wink: